core/web/resolver/mutation.go

```diff
@@ -393,3 +394,39 @@ func (r *Resolver) DeleteBridge(ctx context.Context, args struct {
 
 	return NewDeleteBridgePayload(&bt, nil), nil
 }
+
+func (r *Resolver) CreateP2PKey(ctx context.Context) (*CreateP2PKeyPayloadResolver, error) {
+	if err := authenticateUser(ctx); err != nil {
+		return nil, err
+	}
+
+	key, err := r.App.GetKeyStore().P2P().Create()
+	if err != nil {
```

Comment:
The `ErrLocked` situation also applies here, I'm just not really sure how would be the best way to just wrap this, maybe we just leave it this way?

Reply:
I would need @jkongie input I guess...

Reply:
what's the `ErrLocked` situation? the app shouldn't be allowed to start without unlocking the keystore, so you should be able to assume that it will be unlocked. (Not sure if that helps).

Reply:
Yeah `ErrLocked` will never happen because the unlocking the `Keystore` happens before the HTTP server is brought up.

Reply:
See comment here: https://github.com/smartcontractkit/chainlink/pull/5376#issuecomment-964322507. TLDR: we're going to fix this up in future PR(s) but the gist is we want to surface errors in the payload as much as possible as opposed to the `errors` field in the response.